sources/osgAnimation/CollectAnimationUpdateMorphCallbackVisitor.js

```diff
@@ -0,0 +1,81 @@
+define( [
+    'osg/Notify',
+    'osg/Utils',
+    'osg/NodeVisitor',
+    'osgAnimation/AnimationUpdateCallback',
+    'osgAnimation/UpdateMorph',
+    'osgAnimation/RigGeometry',
+    'osgAnimation/MorphGeometry'
+], function ( Notify, MACROUTILS, NodeVisitor, AnimationUpdateCallback, UpdateMorph, RigGeometry, MorphGeometry ) {
+    'use strict';
+
+    // search into a subgraph all updateMorph
+    var CollectAnimationUpdateMorphCallbackVisitor = function () {
+        NodeVisitor.call( this );
+        this._animationUpdateCallback = {};
+        this._callbackUniqIdMap = {};
+    };
+
+    CollectAnimationUpdateMorphCallbackVisitor.prototype = MACROUTILS.objectInherit( NodeVisitor.prototype, {
```

Comment:
This collector collect UpdateMorph callBacks , find each target on morphGeometry and builds targets name (like cube.0 or sphere.1) . We use it in BasicAnimationManager to link channel with float target